src/Sylius/Component/Promotion/Generator/CouponGenerator.php

```diff
@@ -68,13 +72,13 @@ public function generate(PromotionInterface $promotion, Instruction $instruction
     /**
      * {@inheritdoc}
      */
-    public function generateUniqueCode()
+    public function generateUniqueCode($codeLength)
     {
         $code = null;
 
         do {
             $hash = sha1(microtime(true));
-            $code = strtoupper(substr($hash, mt_rand(0, 33), 6));
+            $code = strtoupper(substr($hash, 0, $codeLength));
```

Comment:
If code length is more than 40, then generated code will still have only 40 characters.